ciphers/src/subtle/rust_crypto/xchacha20poly1305_ietf.rs

```diff
@@ -13,6 +14,58 @@ pub const TAG_LEN: usize = typenum2const! { <AeadImpl as AeadCore>::TagSize };
 /// The nonce length is 24 bytes or 192 bits.
 pub const NONCE_LEN: usize = typenum2const! { <AeadImpl as AeadCore>::NonceSize };
 
+pub struct XChaCha20Poly1305;
+
+impl Aead<KEY_LEN, NONCE_LEN, TAG_LEN> for XChaCha20Poly1305 {
+    fn encrypt(
+        &self,
+        ciphertext: &mut [u8],
+        key: &[u8; KEY_LEN],
+        nonce: &[u8; NONCE_LEN],
+        ad: &[u8],
+        plaintext: &[u8],
+    ) -> Result<(), AeadError> {
+        let nonce = GenericArray::from_slice(nonce);
+        let (n, ct_mac) = ciphertext.split_at_mut(NONCE_LEN);
```

Comment:
This may panic if `ciphertext.len() < NONCE_LEN`. Is this always checked on the outside or should it get checked here?